gtests/test_nopython_callprofilerserializer.cpp

```diff
@@ -0,0 +1,554 @@
+                int uniqueNumber = std::stoi(uniqueNumberString);
+                childrenUniqueNumbers.insert(uniqueNumber);
+            }
+
+            for (const auto & child : node->children())
+            {
+                int uniqueNumber = node_to_unique_number[child.get()];
+                EXPECT_TRUE(childrenUniqueNumbers.find(uniqueNumber) != childrenUniqueNumbers.end());
+                childrenUniqueNumbers.erase(uniqueNumber);
+            }
+        }
+    }
+}
+
+std::vector<std::vector<std::string>> CallProfilerTest::split_nodes_info(const std::vector<std::string> & lines)
+{
+    int numOfLines = lines.size();
+    // Split the nodes serialization into a vector of strings.
+    std::vector<std::vector<std::string>> nodesInfo;
+    auto lineNodesBegin = R"(        "nodes": [{)";
+    int curNodeInfoBegin = std::find(lines.begin(), lines.end(), lineNodesBegin) - lines.begin();
+    int curNodeInfoEnd = 0;
+
+    while (curNodeInfoEnd < numOfLines - 4)
+    {
+        std::vector<std::string> nodeInfo;
+        curNodeInfoEnd = std::find(lines.begin() + curNodeInfoBegin, lines.end(), R"(            },)") - lines.begin();
+        if (curNodeInfoEnd == numOfLines)
+        {
+            curNodeInfoEnd = numOfLines - 4;
+            // Expect the last node has no trailing comma.
+            EXPECT_NE(lines[curNodeInfoEnd].back(), ',');
+        }
+
+        // Unify the first line and last line of the node to make the check easier.
+        nodeInfo.push_back(R"(            {)");
+        for (int i = curNodeInfoBegin + 1; i <= curNodeInfoEnd; ++i)
+        {
+            if (i == numOfLines - 4)
+                nodeInfo.push_back(lines[i] + ",");
+            else
+                nodeInfo.push_back(lines[i]);
+        }
+        nodesInfo.push_back(nodeInfo);
+        curNodeInfoBegin = curNodeInfoEnd + 1;
+    }
+    return nodesInfo;
+}
+
+void CallProfilerTest::BFS_radix_tree(const RadixTreeNode<CallerProfile> * node, node_to_number_map_type & node_to_unique_number, number_to_node_map_type & unique_number_to_node)
+{
+    // BFS the radix tree and number the nodes.
+    std::queue<const RadixTreeNode<CallerProfile> *> nodes_buffer;
+    nodes_buffer.push(node);
+    node_to_unique_number[node] = -1;
+    unique_number_to_node[-1] = node;
+    int unique_node_number = -1;
+
+    while (!nodes_buffer.empty())
+    {
+        const int nodes_buffer_size = nodes_buffer.size();
+        for (int i = 0; i < nodes_buffer_size; ++i)
+        {
+            const RadixTreeNode<CallerProfile> * cur_node = nodes_buffer.front();
+            nodes_buffer.pop();
+            for (const auto & child : cur_node->children())
+            {
+                nodes_buffer.push(child.get());
+                // Store the key and value in the two hash maps.
+                node_to_unique_number[child.get()] = ++unique_node_number;
+                unique_number_to_node[unique_node_number] = child.get();
+            }
+        }
+    }
+}
```

Comment:
* I check the format of the serialization result, including trailing comma ,newline and indent.
* I BFS the radixtree again, and check the correctness of the information of the nodes, and the relationship among nodes.

Reply:
The testing code here is overly complex.  I think they will break very soon as we move forward the development.  At that time, don't hesitate to remove the tests and remake them.

I think the effective form of such tests will be in Python after #344 is done.  Let's leave the tests here for now.

Reply:
I dislike to use names like `BFS` which is not intuitive from high level.  The naming style mixes `CAPITAL` with `snake_case` and looks bad too.  But let's leave it here for now.

Reply:
> I dislike to use names like `BFS` which is not intuitive from high level. The naming style mixes `CAPITAL` with `snake_case` and looks bad too. But let's leave it here for now.

I'm curious whether there is a unified naming style/convention for modmesh.